cloudant.js

```diff
@@ -43,17 +43,26 @@ function Cloudant(options, callback) {
     }
   }
 
-  if (theurl.match(/^http:/)) {
+  if (creds.outUrl.match(/^http:/)) {
     options.https = false;
   }
 
   debug('Creating Cloudant client with options: %j', options);
+  // If IAM key found in VCAP, add IAM plugin
+  if (creds && creds.iamApiKey) {
```

Comment:
In `client.js` the default `cookieauth` plugin is added. Do you think it makes sense to align the approaches a bit more and add the (default) `cookieauth` here in a similar way to IAM? I'm not sure how much pain that would cause, but it might make it easier to follow if defaults are added in the same way and place.

Reply:
@ricellis I think they are both about the same.  I can move the default cookie auth to `cloudant.js`.